VirtualServer.cpp

```diff
@@ -222,8 +225,8 @@ VirtualServer::ReturnCode VirtualServer::processGET(Connection& clientConnection
 
         return RC_IN_PROGRESS;
     }
-
-    if (stat(location.getIndex().c_str(), &buf) == 0
+    const std::string absoluteIndexPath = targetRepresentationURI + "/" + location.getIndex();
```

Comment:
미묘한 차이지만 이렇게 바꿔야 올바른 의도로 동작할 것 같습니다.

```suggestion
    const std::string absoluteIndexPath = location.getRoot() + "/" + location.getIndex();
```

`targetRepresentationURI`는 요청이 URI로 지목한 파일의 절대 경로를 의미합니다.
따라서 지목한 파일의 절대 경로 뒤에 index 파일을 붙이면 요청한 경로가 location의 route와 동일하지 않은 이상 오작동 할 것 같습니다.

Reply:
```
req : http://location:80/put_test/test1
location route : /put_test
location root : ~/webserve/YoupiBanane/nop
```

위와 같은 환경이면  http://location:80/put_test ->  ~/webserve/YoupiBanane/nop가 되고
그 뒤에 `test1`이라는 경로를 붙여줘야해서 `targetRepresentationURI`를 사용한겁니다. 
오늘 나오시면 이거 관련해서 논의해보면 좋을거 같아요

Reply:
이 부분은 제가 index 지시자에 대해 잘못 알고 있었습니다.
모송님이 구현하신 것처럼 targetRepresentationURI에서 index 지시자에 적힌 파일을 찾는게 올바른 동작인 것을 확인했습니다.
고생하셨습니다!